### PR TITLE
App start refactoring (EXPOSUREAPP-12787)

### PR DESCRIPTION
**Pull request by mtwalli**
## Changes
- Refactor App start logic to use Dagger Set injection
- Move to `runTest` completely in unit tests 

## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12787

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ApplicationComponent.kt [hunk 1]
```diff
@@ -125,6 +128,8 @@ interface ApplicationComponent : AndroidInjector<CoronaWarnApplication> {
 
     @AppScope val appScope: AppCoroutineScope
 
+    val initializers: Set<@JvmSuppressWildcards Initializer>
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ApplicationComponent.kt:131]: Why is this needed?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ApplicationComponent.kt:131]: this is leftover , I was calling it from 
```
AppInjector.init(this).let { compPreview ->}
```

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt [hunk 2]
```diff
@@ -65,49 +35,15 @@ class CoronaWarnApplication : Application(), HasAndroidInjector {
...
-    @Inject lateinit var cclConfigurationUpdaterScheduler: CclConfigurationUpdateScheduler
-    @Inject lateinit var familyTestResultRetrievalScheduler: FamilyTestResultRetrievalScheduler
-    @Inject lateinit var dccValidityStateChangeObserver: DccValidityStateChangeObserver
-    @Inject lateinit var dccRevocationUpdateScheduler: DccRevocationUpdateScheduler
-
-    @AppScope
-    @Inject lateinit var appScope: CoroutineScope
-
+    @Inject lateinit var foregroundState: ForegroundState
+    @Inject lateinit var initializers: Set<@JvmSuppressWildcards Initializer>
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt:41]: would not `Lazy` be better ? afaik Provider creates new instances every time `get` is called

**Comment by mtwalli** [comment]
> May we get some details on this PR, please?

It is a refactoring PR to simplify many entry points when the App is started and make adding new logic that require initialization on app start easier (nothing that concerns end user)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt]: But we call them only once inside this class, so i don't really see the benefit of using Lazy instead of Provider.
Btw. most of them are Singletons anyway, so Dagger ensures to create only 1 instance overall.

[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt]: Ok, I don't think either way will have a big impact even direct injection , I changed it anyway

[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt]: Sure, it won't have a big impact, but it's just a little optimization.

**Review by BMItr — APPROVED** [review]
I will miss the overloaded ApplicationClass 😄 
Beside of this, very nice refactoring and cleanup!
Thx 👌

**Review by chiljamgossow — APPROVED** [review]
Works as far as I can see
